Applications/Spire/Source/Charting/LocalChartModel.cpp

```diff
@@ -42,6 +42,24 @@ QtPromise<std::vector<Candlestick>> LocalChartModel::load(ChartValue first,
   });
 }
 
+void LocalChartModel::store(const std::vector<Candlestick>& candlesticks) {
+  if(candlesticks.empty()) {
+    return;
+  }
+  auto first = std::lower_bound(m_candlesticks.begin(),
+    m_candlesticks.end(), candlesticks.front().GetStart(), [] (
+        const auto& index, const auto& value) {
+      return index.GetStart() < value;
+    });
+  auto last = std::lower_bound(m_candlesticks.begin(),
+    m_candlesticks.end(), candlesticks.back().GetEnd(), [] (const auto& index,
+        const auto& value) {
+      return index.GetStart() < value;
+    });
+  auto index = m_candlesticks.erase(first, last);
+  m_candlesticks.insert(index, candlesticks.begin(), candlesticks.end());
```

Comment:
Unfortunately you don't know that you can just insert the candlesticks like this.

Consider LocalModel contains the following: [1, 4, 5, 6, 10]

And you want to insert: [3, 7, 8, 9]

Then the final list should be: [1, 3, 4, 5, 6, 7, 8, 9, 10].

You will need to merge the two sorted ranges together:

Review the following and add an appropriate unit test:

https://en.cppreference.com/w/cpp/algorithm/merge

Reply:
I found that std::merge won't remove duplicates but std::set_union will. The last issue is finding out how to avoid an unnecessary assignment. Currently I have:

`  auto c = std::vector<Candlestick>();
  c.reserve(m_candlesticks.size() + candlesticks.size());
  std::set_union(m_candlesticks.begin(), m_candlesticks.end(), candlesticks.begin(),
    candlesticks.end(), std::back_inserter(c),
    [=] (auto first, auto second) {
      return first.GetStart() < second.GetStart();
    });
  m_candlesticks = std::move(c);`

But it is possible to just insert the values into m_candlesticks as they're evaluated?